### PR TITLE
feat: lemmas analogous to fderiv_within_of_isOpen

### PR DESCRIPTION
**Pull request by grunweg**
- from the project towards Sard's theorem

### CODE REVIEW

File: Mathlib/Analysis/Calculus/FDeriv/Basic.lean [hunk 1]
```diff
@@ -409,6 +409,14 @@ theorem hasFDerivWithinAt_univ : HasFDerivWithinAt f f' univ x ↔ HasFDerivAt f
 alias ⟨HasFDerivWithinAt.hasFDerivAt_of_univ, _⟩ := hasFDerivWithinAt_univ
 #align has_fderiv_within_at.has_fderiv_at_of_univ HasFDerivWithinAt.hasFDerivAt_of_univ
 
+theorem hasFDerivWithin_of_mem_nhds (h : s ∈ 𝓝 x) :
```
Comment (sgouezel) [inline Mathlib/Analysis/Calculus/FDeriv/Basic.lean:412]:
```suggestion
theorem hasFDerivWithinAt_of_mem_nhds (h : s ∈ 𝓝 x) :
```

File: Mathlib/Analysis/Calculus/FDeriv/Basic.lean [hunk 2]
```diff
@@ -409,6 +409,14 @@ theorem hasFDerivWithinAt_univ : HasFDerivWithinAt f f' univ x ↔ HasFDerivAt f
...
+    HasFDerivWithinAt f f' s x ↔ HasFDerivAt f f' x := by
+  rw [HasFDerivAt, HasFDerivWithinAt, nhdsWithin_eq_nhds.mpr h]
+
+lemma hasFDerivWithinAt_of_open (h : IsOpen s) (hx : x ∈ s) :
```
Comment (sgouezel) [inline Mathlib/Analysis/Calculus/FDeriv/Basic.lean:416]:
```suggestion
lemma hasFDerivWithinAt_of_isOpen (h : IsOpen s) (hx : x ∈ s) :
```

File: Mathlib/Geometry/Manifold/MFDeriv.lean [hunk 3]
```diff
@@ -639,6 +639,18 @@ theorem mfderivWithin_inter (ht : t ∈ 𝓝 x) :
     fderivWithin_inter (extChartAt_preimage_mem_nhds I x ht)]
 #align mfderiv_within_inter mfderivWithin_inter
 
+theorem mfderivWithin_of_mem_nhds (h : s ∈ 𝓝 x) : mfderivWithin I I' f s x = mfderiv I I' f x := by
+  rw [← mfderivWithin_univ, ← univ_inter s, mfderivWithin_inter h]
+
+lemma mfderivWithin_of_open (hs : IsOpen s) (hx : x ∈ s) :
```
Comment (sgouezel) [inline Mathlib/Geometry/Manifold/MFDeriv.lean:645]:
```suggestion
lemma mfderivWithin_of_isOpen (hs : IsOpen s) (hx : x ∈ s) :
```

**Comment by grunweg** [comment]
Thanks; all addressed.
I was mirroring the lemma `fderiv_within_of_open` - I take it should be renamed; I've pushed a commit accordingly.

**Comment by sgouezel** [comment]
bors r+
maintainer merge
Thanks!

**Pull request merged**